src/Components/Web/src/Image/Image.cs

```diff
@@ -0,0 +1,267 @@
+ *
+ * <img class="blazor-image @GetCssClass()"
+ *      data-state=@(_isLoading ? "loading" : _hasError ? "error" : null)
+ *      @ref="Element" @attributes="AdditionalAttributes" />
+ *
+ */
+/// <summary>
+/// A component that efficiently renders images from non-HTTP sources like byte arrays.
+/// </summary>
+public partial class Image : IComponent, IHandleAfterRender, IAsyncDisposable
+{
+    private RenderHandle _renderHandle;
+    private string? _currentObjectUrl;
+    private bool _hasError;
+    private bool _isDisposed;
+    private bool _initialized;
+    private bool _hasPendingRender;
+    private string? _activeCacheKey;
+    private ImageSource? _currentSource;
+    private CancellationTokenSource? _loadCts;
+    private bool IsLoading => _currentSource != null && string.IsNullOrEmpty(_currentObjectUrl) && !_hasError;
+
+    private bool IsInteractive => _renderHandle.IsInitialized &&
+                                _renderHandle.RendererInfo.IsInteractive;
+
+    [DisallowNull] private ElementReference? Element { get; set; }
+
+    [Inject] private IJSRuntime JSRuntime { get; set; } = default!;
+
+    [Inject] private ILogger<Image> Logger { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets the source for the image.
+    /// </summary>
+    [Parameter] public ImageSource? Source { get; set; }
+
+    /// <summary>
+    /// Gets or sets the attributes for the image.
+    /// </summary>
+    [Parameter(CaptureUnmatchedValues = true)] public Dictionary<string, object>? AdditionalAttributes { get; set; }
+
+    void IComponent.Attach(RenderHandle renderHandle)
+    {
+        if (_renderHandle.IsInitialized)
+        {
+            throw new InvalidOperationException("Component is already attached to a render handle.");
+        }
+        _renderHandle = renderHandle;
+    }
+
+    Task IComponent.SetParametersAsync(ParameterView parameters)
+    {
+        var previousSource = Source;
+
+        // Set component parameters
+        parameters.SetParameterProperties(this);
+
+        // Initialize on first parameters set
+        if (!_initialized)
+        {
+            Render();
+            _initialized = true;
+            return Task.CompletedTask;
+        }
+
+        if (Source != null && !string.Equals(previousSource?.CacheKey, Source.CacheKey, StringComparison.Ordinal))
+        {
+            Render();
+        }
+
+        return Task.CompletedTask;
+    }
+
+    async Task IHandleAfterRender.OnAfterRenderAsync()
+    {
+        if (!IsInteractive || Source == null)
+        {
+            return;
+        }
+
+        if (_currentSource != null && string.Equals(_currentSource.CacheKey, Source.CacheKey, StringComparison.Ordinal))
+        {
+            return;
+        }
+
+        // Cancel any in-progress load operation
+        try { _loadCts?.Cancel(); } catch { }
+        _loadCts?.Dispose();
+        _loadCts = new CancellationTokenSource();
+        var token = _loadCts.Token;
```

Comment:
Move this to a separate helper method? (CancelPreviousLoad, ResetCancellationToken) and so on).
Also, don't do the try..catch in one line.